### PR TITLE
RC-33186 Added support for addon criticality in terraform.

### PR DESCRIPTION
**Pull request by rafay-vivek**

### CODE REVIEW

**Review by saisrikar-rafay — APPROVED** [review]
LGTM

File: examples/resources/rafay_blueprint/resource.tf [hunk 1]
```diff
@@ -135,10 +135,9 @@ resource "rafay_blueprint" "blueprint" {
       name = "addon1"
       version = "v0"
     }
-    custom_addons {
-      depends_on = ["addon1"]
-      name = "addon2"
-      version = "v0"
+    components_criticality {
```
Comment (krishnarafay) [inline examples/resources/rafay_blueprint/resource.tf:138]: Don't delete `custom_addons` ( that is needed as it contains how to add depends_on)..

**Review by saisrikar-rafay — COMMENTED** [review]
https://github.com/RafaySystems/terraform-provider-rafay/pull/532/files#r1537046600

File: examples/resources/rafay_blueprint/resource.tf (hunk 1 above)
Comment (rafay-vivek) [reply examples/resources/rafay_blueprint/resource.tf:143]: Addressed your comment @krishnarafay

**Review by saisrikar-rafay — APPROVED** [review]

**Review by krishnarafay — APPROVED** [review]

**Comment by krishnarafay** [comment]
NOTE: Update go.mod before merge.

File: rafay/resource_blueprint.go [hunk 2]
```diff
@@ -1625,6 +1678,46 @@ func flattenResources(in *commonpb.ResourceRequirements, p []interface{}) []inte
...
+			obj = p[i].(map[string]interface{})
+		}
+		if len(in.Name) > 0 {
+			obj["name"] = in.Name
+		}
+		if len(in.Version) > 0 {
+			obj["version"] = in.Version
+		}
+		if len(in.ComponentType) > 0 {
+			obj["componentType"] = in.Version
```
Comment (janardhan-rafay) [inline rafay/resource_blueprint.go:1698]: this has to be `in.ComponentType` right?

**Review by janardhan-rafay — APPROVED** [review]

**Pull request merged**